arch/sim/src/sim/up_hostmemory.c

```diff
@@ -63,8 +63,14 @@ void *host_alloc_heap(size_t sz)
 {
   void *p;
 
+#if defined(CONFIG_HOST_MACOS) && defined(CONFIG_HOST_ARM64)
+  p = mmap(NULL, sz, PROT_READ | PROT_WRITE,
```

Comment:
macos don't allow we mmap the memory with execute permission?

Reply:
That's a known issue reported in MacOS 11.2 on M1 silicon.
See following discussion: https://developer.apple.com/forums/thread/672804